stable/openldap/templates/deployment.yaml

```diff
@@ -7,32 +7,40 @@ metadata:
     chart: {{ template "openldap.chart" . }}
     release: {{ .Release.Name }}
     heritage: {{ .Release.Service }}
+{{- if .Values.extraLabels }}
+{{ toYaml .Values.extraLabels | indent 4 }}
+{{- end }}
 spec:
-  replicas: {{ .Values.replicas }}
+  replicas: {{ .Values.replicaCount }}
   selector:
     matchLabels:
       app: {{ template "openldap.name" . }}
       release: {{ .Release.Name }}
   template:
     metadata:
+      annotations:
+        checksum/configmap-env: {{ include (print $.Template.BasePath "/configmap-env.yaml") . | sha256sum }}
+{{- if .Values.customLdifFiles}}
+        checksum/configmap-customldif: {{ include (print $.Template.BasePath "/configmap-customldif.yaml") . | sha256sum }}
+{{- end }}
       labels:
         app: {{ template "openldap.name" . }}
         release: {{ .Release.Name }}
     spec:
       containers:
-        - name: {{ .Values.containerName | quote }}
+        - name: {{ .Chart.Name }}
           image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
-          imagePullPolicy: {{ .Values.image.pullPolicy | quote }}
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
           ports:
             - name: ldap-port
-              containerPort: 389
+              containerPort: {{ .Values.service.ldapPort }}
```

Comment:
We have explicitly hard coded these here since there is no way to configure the ports in the container like this. {{service.ldapPort}} defines the external port as opposed to the internal port for the container.